test/spanner.ts

```diff
@@ -5075,6 +5076,23 @@ describe('Spanner with mock server', () => {
       done();
     });
   });
+
+  describe('session-factory', () => {
+    after(() => {
+      process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS = 'false';
+    });
+
+    it('should not throw error when enabling env GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS after client initialization', done => {
+      const database = newTestDatabase();
+      // enable env after database creation
+      process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS = 'true';
+      const sessionFactory = database.sessionFactory_ as SessionFactory;
+      sessionFactory.getSession((err, _) => {
+        assert.ifError(err);
+        done();
+      });
+    });
+  });
```

Comment:
This should go to database class

Reply:
since, we are getting sessionFactory as an instance of FakeSessionFactory instead of SessionFactory, 
in FakeSessionFactory, we are unable to use in-memory variable - https://github.com/googleapis/nodejs-spanner/blob/get-session-class/test/database.ts#L133-#L152 instead we are dynamically fetching the env variable configuration using `process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS `

Hence, even though we are enabling GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS after client initialization, still getSession is returning FakeMultiplexedSession which is not an expected behavior (expected behavior is to return FakeSession if enabling GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS after initialization)

Reply:
We need to keep this test in spanner test, so that we can test the desired behavior after changing the GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS after client initialization